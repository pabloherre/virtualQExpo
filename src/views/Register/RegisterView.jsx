import React, { Component } from 'react';
import { connect } from 'react-redux';

import { RoundedButton } from '../../common/buttons';
import { TextInput } from '../../common/inputs';
import { AsyncStorage, Image, SafeAreaView, View } from 'react-native';
import { setUser } from '../../modules/auth/Auth.actions';

import logo from '../../../assets/images/logo1.jpg';
import { safeArea } from '../../styles/common.styles';

class RegisterView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      lastName: '',
      email: '',
      password: ''
    };
  }

  handleChange = (value, field) => {
    this.setState({ [field]: value });
  };

  handleRegister = async () => {
    try {
      await AsyncStorage.setItem('user', JSON.stringify(this.state));
      this.props.setUser(this.state);
      this.props.navigation.navigate('Appointments');
    } catch (e) {
      console.log(e);
    }
  };

  render() {
    const { name, lastName, email, password } = this.state;
    return (
      <SafeAreaView style={safeArea}>
        <View style={{ marginTop: 100 }}>
          <Image source={logo} style={{ alignSelf: 'center' }} />
          <View style={{ marginBottom: 20 }}>
            <TextInput value={name} onChangeText={text => this.handleChange(text, 'name')} label="Nombre" />
            <TextInput value={lastName} onChangeText={text => this.handleChange(text, 'lastName')} label="Apellido" />
            <TextInput value={email} onChangeText={text => this.handleChange(text, 'email')} label="Email" />
            <TextInput value={password} secureTextEntry onChangeText={text => this.handleChange(text, 'password')} label="Password" />
          </View>
          <RoundedButton label="REGISTRARME" onPress={this.handleRegister} />
          {/*<Button*/}
          {/*  onPress={() => {*/}
          {/*    this.props.navigation.navigate('Appointments');*/}
          {/*  }}*/}
          {/*  title={'Appointments'}*/}
          {/*/>*/}
        </View>
      </SafeAreaView>
    );
  }
}

const mapDispatchToProps = {
  setUser
};

export default connect(null, mapDispatchToProps)(RegisterView);
