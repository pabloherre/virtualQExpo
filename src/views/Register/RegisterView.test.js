import React from 'react';
import { RegisterView } from './RegisterView';
import { shallow, mount } from 'enzyme';
import AsyncStorage from '@react-native-community/async-storage';

const mockProps = {
  setUser: jest.fn(),
  navigation: {
    navigate: jest.fn()
  }
};

describe('<RegisterView />', () => {
  it('should render correctly', async () => {
    const wrapper = mount(<RegisterView />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should update email state when user types in', async () => {
    const wrapper = shallow(<RegisterView />);
    wrapper.find('TextInput[label="Nombre"]').props().onChangeText('testName');
    expect(wrapper.state('name')).toBe('testName');
  });

  it('should update password state when user types in', async () => {
    const wrapper = shallow(<RegisterView />);
    wrapper.find('TextInput[label="Apellido"]').props().onChangeText('testLastName');
    expect(wrapper.state('lastName')).toBe('testLastName');
  });
  it('should update email state when user types in', async () => {
    const wrapper = shallow(<RegisterView />);
    wrapper.find('TextInput[label="Email"]').props().onChangeText('testEmail');
    expect(wrapper.state('email')).toBe('testEmail');
  });

  it('should update password state when user types in', async () => {
    const wrapper = shallow(<RegisterView />);
    wrapper.find('TextInput[label="Password"]').props().onChangeText('testPassword');
    expect(wrapper.state('password')).toBe('testPassword');
  });

  it('should set user in async storage when register', async () => {
    const wrapper = shallow(<RegisterView {...mockProps} />);

    await wrapper.instance().handleRegister();

    expect(AsyncStorage.setItem).toHaveBeenCalled();
    expect(wrapper.instance().props.setUser).toHaveBeenCalled();
    expect(wrapper.instance().props.navigation.navigate).toHaveBeenCalled();
  });
});
