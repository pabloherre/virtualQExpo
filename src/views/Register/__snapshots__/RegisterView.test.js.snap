// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterView /> should render correctly 1`] = `
<RegisterView>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
        "paddingHorizontal": 40,
        "paddingVertical": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "paddingHorizontal": 40,
          "paddingVertical": 20,
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": 100,
          }
        }
      >
        <View
          style={
            Object {
              "marginTop": 100,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            />
          </Image>
          <View
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "marginBottom": 20,
                }
              }
            >
              <TextInput
                label="Nombre"
                onChangeText={[Function]}
                value=""
              >
                <View
                  style={
                    Object {
                      "paddingBottom": 10,
                      "paddingTop": 10,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "paddingBottom": 10,
                        "paddingTop": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#0db4e4",
                          "fontSize": 20,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#0db4e4",
                            "fontSize": 20,
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Nombre
                      </Text>
                    </Text>
                    <Component
                      allowFontScaling={true}
                      onChangeText={[Function]}
                      rejectResponderTermination={true}
                      style={
                        Object {
                          "borderBottomColor": "#0db4e4",
                          "borderBottomWidth": 1,
                          "fontSize": 20,
                          "height": 40,
                        }
                      }
                      underlineColorAndroid="transparent"
                      value=""
                    >
                      <TextInput
                        allowFontScaling={true}
                        onChangeText={[Function]}
                        rejectResponderTermination={true}
                        style={
                          Object {
                            "borderBottomColor": "#0db4e4",
                            "borderBottomWidth": 1,
                            "fontSize": 20,
                            "height": 40,
                          }
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </Component>
                  </View>
                </View>
              </TextInput>
              <TextInput
                label="Apellido"
                onChangeText={[Function]}
                value=""
              >
                <View
                  style={
                    Object {
                      "paddingBottom": 10,
                      "paddingTop": 10,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "paddingBottom": 10,
                        "paddingTop": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#0db4e4",
                          "fontSize": 20,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#0db4e4",
                            "fontSize": 20,
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Apellido
                      </Text>
                    </Text>
                    <Component
                      allowFontScaling={true}
                      onChangeText={[Function]}
                      rejectResponderTermination={true}
                      style={
                        Object {
                          "borderBottomColor": "#0db4e4",
                          "borderBottomWidth": 1,
                          "fontSize": 20,
                          "height": 40,
                        }
                      }
                      underlineColorAndroid="transparent"
                      value=""
                    >
                      <TextInput
                        allowFontScaling={true}
                        onChangeText={[Function]}
                        rejectResponderTermination={true}
                        style={
                          Object {
                            "borderBottomColor": "#0db4e4",
                            "borderBottomWidth": 1,
                            "fontSize": 20,
                            "height": 40,
                          }
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </Component>
                  </View>
                </View>
              </TextInput>
              <TextInput
                label="Email"
                onChangeText={[Function]}
                value=""
              >
                <View
                  style={
                    Object {
                      "paddingBottom": 10,
                      "paddingTop": 10,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "paddingBottom": 10,
                        "paddingTop": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#0db4e4",
                          "fontSize": 20,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#0db4e4",
                            "fontSize": 20,
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Email
                      </Text>
                    </Text>
                    <Component
                      allowFontScaling={true}
                      onChangeText={[Function]}
                      rejectResponderTermination={true}
                      style={
                        Object {
                          "borderBottomColor": "#0db4e4",
                          "borderBottomWidth": 1,
                          "fontSize": 20,
                          "height": 40,
                        }
                      }
                      underlineColorAndroid="transparent"
                      value=""
                    >
                      <TextInput
                        allowFontScaling={true}
                        onChangeText={[Function]}
                        rejectResponderTermination={true}
                        style={
                          Object {
                            "borderBottomColor": "#0db4e4",
                            "borderBottomWidth": 1,
                            "fontSize": 20,
                            "height": 40,
                          }
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </Component>
                  </View>
                </View>
              </TextInput>
              <TextInput
                label="Password"
                onChangeText={[Function]}
                secureTextEntry={true}
                value=""
              >
                <View
                  style={
                    Object {
                      "paddingBottom": 10,
                      "paddingTop": 10,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "paddingBottom": 10,
                        "paddingTop": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#0db4e4",
                          "fontSize": 20,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#0db4e4",
                            "fontSize": 20,
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Password
                      </Text>
                    </Text>
                    <Component
                      allowFontScaling={true}
                      onChangeText={[Function]}
                      rejectResponderTermination={true}
                      secureTextEntry={true}
                      style={
                        Object {
                          "borderBottomColor": "#0db4e4",
                          "borderBottomWidth": 1,
                          "fontSize": 20,
                          "height": 40,
                        }
                      }
                      underlineColorAndroid="transparent"
                      value=""
                    >
                      <TextInput
                        allowFontScaling={true}
                        onChangeText={[Function]}
                        rejectResponderTermination={true}
                        secureTextEntry={true}
                        style={
                          Object {
                            "borderBottomColor": "#0db4e4",
                            "borderBottomWidth": 1,
                            "fontSize": 20,
                            "height": 40,
                          }
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </Component>
                  </View>
                </View>
              </TextInput>
            </View>
          </View>
          <RoundedButton
            buttonStyle={Object {}}
            label="REGISTRARME"
            onPress={[Function]}
            textStyle={Object {}}
          >
            <ForwardRef
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f77027",
                  "borderRadius": 30,
                  "flexDirection": "row",
                  "height": 60,
                  "justifyContent": "space-around",
                  "paddingHorizontal": 20,
                  "paddingVertical": 10,
                }
              }
            >
              <TouchableOpacity
                hostRef={null}
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#f77027",
                    "borderRadius": 30,
                    "flexDirection": "row",
                    "height": 60,
                    "justifyContent": "space-around",
                    "paddingHorizontal": 20,
                    "paddingVertical": 10,
                  }
                }
              >
                <ForwardRef(AnimatedComponentWrapper)
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#f77027",
                        "borderRadius": 30,
                        "flexDirection": "row",
                        "height": 60,
                        "justifyContent": "space-around",
                        "paddingHorizontal": 20,
                        "paddingVertical": 10,
                      },
                      Object {
                        "opacity": 1,
                      },
                    ]
                  }
                >
                  <AnimatedComponent
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#f77027",
                          "borderRadius": 30,
                          "flexDirection": "row",
                          "height": 60,
                          "justifyContent": "space-around",
                          "paddingHorizontal": 20,
                          "paddingVertical": 10,
                        },
                        Object {
                          "opacity": 1,
                        },
                      ]
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#f77027",
                          "borderRadius": 30,
                          "flexDirection": "row",
                          "height": 60,
                          "justifyContent": "space-around",
                          "opacity": 1,
                          "paddingHorizontal": 20,
                          "paddingVertical": 10,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#f77027",
                            "borderRadius": 30,
                            "flexDirection": "row",
                            "height": 60,
                            "justifyContent": "space-around",
                            "opacity": 1,
                            "paddingHorizontal": 20,
                            "paddingVertical": 10,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "white",
                              "fontSize": 20,
                              "paddingHorizontal": 10,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 20,
                                "paddingHorizontal": 10,
                              }
                            }
                          >
                            REGISTRARME
                          </Text>
                        </Text>
                        <PressabilityDebugView
                          color="cyan"
                        />
                      </View>
                    </View>
                  </AnimatedComponent>
                </ForwardRef(AnimatedComponentWrapper)>
              </TouchableOpacity>
            </ForwardRef>
          </RoundedButton>
        </View>
      </View>
    </View>
  </View>
</RegisterView>
`;
